mailtrap/api/base.py

```
@@ -0,0 +1,71 @@
+            return data
+
+        self._handle_failed_response(response)
+
+    @staticmethod
+    def _handle_failed_response(response: Response) -> NoReturn:
+        status_code = response.status_code
+
+        try:
+            data = response.json()
+        except ValueError:
+            raise APIError(status_code, errors=["Unknown Error"])
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)